Oni species

pubby opened the pull request:
Hi friends. Here is a finalized version of my magic system that I've put on a new species.

**The gist of the species is:**

- You don't learn spells from spell books, instead you 'discover' spells by gaining experience. The formula is 1 random spell every 2 levels, up to level 26.
- Because of this, you only discover a few (13) spells per game.
- The only spell school is spellcasting. All spells are "spellcasting" type.

This results in very unique spell sets that have a toolbox feel.

**Why is it on a new species instead of on an existing one?**

Existing species aptitudes did not match up with the balance I wanted, so creating a new species made balancing easier.

**What are the aptitudes?**

Notably: +0 fighting, +1 maces, +0 UC, -1 Armour, -1 Dodging, +2 Invo, -4 Spellcasting

Originally fighting was +1 and weapons were -1, but I found reversing that worked better for non-book starts.

**Should there be an additional gimmick?**

I'm still hoping for mainline to consider this, so I'm hesitant about adding more gimmicks. It's unfortunate, but mainline devs perceive such things as bloat.

For now, I think the magic system is enough.

pubby committed:
New species: Oni

a.k.a Japanese ogre demon things.

Oni magic works differently than other species. Instead of learning spells
through books, Oni learn spells when they level up. Additionally,
Oni cast using spellcasting only, which replaces individual spell
schools for them.

Formula: Gain 1 random spell every 2 XLs, up to level 26.
         (A total of 13 spells per game.)
         Low level spells are gained first, then strong ones.

These two changes focus the gameplay around using a small set of
spells of differing schools. Unlike other casters, your magic toolbox
is very limited, but unique. You might get a single conjuration, a single
summon, a charm, a necromancy, and two hexes, and you have to figure out how
to tactically use them.

The species has fairly flat  aptitudes and a few monstrous innate mutations.
They are hated by holy gods.

Tweak Oni apts

Raising weapon apts makes hybrids more viable, as in some games you'll
get charms instead of conjurations.

Dummy oni monster

Recommend Gl and Tm to Oni

pubby committed:
Remove twisted resurrection reference in Oni

NormalPerson7 commented:
You could branch your `xoni` branch into `xoni-xcrawl` or something so that you can add other things if you really wanted to, remake the PR using that and then not have your mainline crawl PR take changes you want for X-Crawl. That's only a suggestion.

Besides, this has (for now) been approved, so I, and some other people, will do some testing on it to look at minor changes we see fit. Looks good generally though.

(Just one thing: -4 spellcasting apt doesn't really fit thematically imo, even though it's pretty necessary for balance. I'm undecided on what I want done with that.)

pubby committed:
Buff level scaling.

You now get better spells slightly earlier.

pubby committed:
Simplify recommended Oni backgrounds

Other species have 6-8 recommended backgrounds, so Oni should have
that number too!

All recommended Oni backgrounds are book starts, as gaining access to more
spells is really helpful to Oni.

pubby committed:
Buff Oni str

Flavor reasons

NormalPerson7 committed:
Merge branch 'master' into xoni

NormalPerson7 committed:
Wrap some lines

These lines exceeded the maximum line length of 80 characters.
